policy/release/olm.rego

```diff
@@ -0,0 +1,114 @@
+#
+# METADATA
+# description: >-
+#   Checks for Operator Lifecycle Manager (OLM) bundles.
+#
+package policy.release.olm
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.lib
+import data.lib.image
+
+olm_manifestv1 := "operators.operatorframework.io.bundle.manifests.v1"
+
+# METADATA
+# title: Unpinned images in OLM bundle
+# description: >-
+#   Checks the OLM bundle image for the presence of unpinned image references.
+# custom:
+#   short_name: unpinned_references
+#   failure_msg: The %q image reference is not pinned at %s.
+#   solution: >-
+#     Update the OLM bundle replacing the unpinned image reference with pinned image
+#     reference. Pinned image reference contains the image digest.
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	manifestDir := input.image.config.Labels[olm_manifestv1]
+
+	some path, manifest in input.image.files
+
+	# only consider files in the manifest path as determined by the OLM manifest v1 label
+	startswith(path, manifestDir)
+
+	# only consider this API prefix, disregard the version
+	startswith(manifest.apiVersion, "operators.coreos.com/")
+
+	# only consider CSV manifests
+	manifest.kind == "ClusterServiceVersion"
+
+	some i in all_image_ref(manifest)
+	i.ref.digest == "" # unpinned image references have no digest
+
+	result := lib.result_helper_with_term(rego.metadata.chain(), [image.str(i.ref), i.path], image.str(i.ref))
+}
+
+_name(o) := n if {
+	n := o.name
+} else := "unnamed"
+
+# Finds all image references and their locations (paths). Returns all image
+# references (parsed into components) found in locations as specified by:
+# https://github.com/containerbuildsystem/operator-manifest/blob/f24cd9374f5ad9fed04f47701acffa16837d940e/README.md#pull-specifications
```

Comment:
Alternatively, consider adding those to the package docs. We should probably copy from https://github.com/containerbuildsystem/operator-manifest#pull-specifications instead of just linking to it given that we are re-implementing that.